packages/accounts-base/accounts_server.js

```diff
@@ -77,9 +77,10 @@ export class AccountsServer extends AccountsCommon {
     // user expects. The way to make this work in a publish is to do
     // Meteor.find(this.userId).observe and recompute when the user
     // record changes.
-    var currentInvocation = DDP._CurrentInvocation.get();
+    var currentInvocation = DDP._CurrentPublicationInvocation.get();
+    currentInvocation = currentInvocation || DDP._CurrentInvocation.get();
     if (!currentInvocation)
-      throw new Error("Meteor.userId can only be invoked in method calls. Use this.userId in publish functions.");
+      throw new Error("Meteor.userId can only be invoked in method calls or publications.");
```

Comment:
I'll try and generalize this comment a bit as this holds true for methods as well, as far as I can tell.
i.e. `Meteor.user()` won't be reactive on server-side either in a method or a publication.